cmd/agent/common/common_windows.go

```diff
@@ -143,6 +143,27 @@ func CheckAndUpgradeConfig() error {
 }
 
 // ImportRegistryConfig imports settings from Windows registry into datadog.yaml
+//
+// Config settings are placed in the registry by the Windows (MSI) installer.  The
+// registry is used as an intermediate step to hold the config options in between
+// the installer running and the first run of the agent.
+//
+// The agent will only apply these settings on a new install.  A new install is determined
+// by the existence of an API key in the config file (datadog.yaml).  Existence of the API key
+// indicates this is an upgrade, and the settings provided on the command line (via the registry)
+// will be ignored.
+//
+// Lack of an API key is interpreted as a new install.  Take any/all of the options supplied
+// on the command line and apply them to the config, and then write the configuration file
```

Comment:
Perhaps:

```
on the command line and apply them to the config, and overwrite the configuration file
to persist the command line options. The yaml configuration file is the single source of truth,
and thus the registry entries created by the installer are deleted to avoid confusion.
```